### PR TITLE
Add the option to configure the bind password using an environment variable for the LDAP connector.

### PR DESCRIPTION
**Pull request by onkarbhat**
In a k8s deployment that uses Dex, this will allow us to define an environment variable using a k8s secret that contains the bind password instead of showing it in plaintext in a configmap. 

This partially meets requirements in #1099.

Related  issue - https://github.com/dexidp/dex/issues/1158 

cc: @sagikazarmark @bonifaido @danilina-wsib

### CODE REVIEW

File: connector/ldap/ldap.go [hunk 1]
```diff
@@ -237,6 +239,18 @@ func (c *Config) openConnector(logger log.Logger) (*ldapConnector, error) {
...
 
+	if c.BindPW == "" && c.BindPWFromEnv == "" {
+		return nil, fmt.Errorf("ldap: bindPW or bindPWFromEnv are required for the LDAP connector")
+	}
+	bindPW := c.BindPW
+	if c.BindPWFromEnv != "" {
+		if c.BindPW != "" {
+			return nil, fmt.Errorf("ldap: bindPW and bindPWFromEnv are exclusive for the LDAP connector")
+		}
+		bindPW = os.Getenv(c.BindPWFromEnv)
```
Comment (vkamra) [inline connector/ldap/ldap.go:250]: Use `LookupEnv` instead and fail if the environment variable is not present.

Comment (onkarbhat) [reply connector/ldap/ldap.go]: Thanks @vkamra . I've addressed the comment and added a unit test to check for error cases related to LookupEnv.

**Comment by heidemn** [comment]
@onkarbhat have you seen the new config templating feature, coming in the next release?
This should solve your problem already.
[examples/config-tmpl.yaml](https://github.com/dexidp/dex/pull/1965/files#diff-c7f8bd4d261664ee2b089de54d7a767b3a03409af065e9eb51fc95592cc384c0)

Note that the config file is just an example. You can define your own, using all the templating logic you need.

**Comment by sagikazarmark** [comment]
Yeah, I don't think we'll be accepting this one. @onkarbhat please try the new templating feature.

**Comment by onkarbhat** [comment]
Thank you @heidemn @sagikazarmark 👍🏼 .   I believe these are the steps that a user would have to execute in order to configure a password in the LDAP connector config in Dex in a K8s cluster. 

- Create a secret that contains the password.
- Mount the secret's entry as an environment variable in the Dex Pod. Lets call this env as LDAP_BIND_PW_ENV.
- Prepare a config.docker.yaml with the bindPW  set like this: `{{ getenv "LDAP_BIND_PW_ENV" "" }}`
- Store config.docker.yaml in a config map, and mount it at /etc/dex/config.docker.yaml .

**Comment by sagikazarmark** [comment]
That sounds about right. I'd probably consider using [bank-vaults](https://github.com/banzaicloud/bank-vaults) or [kube-secrets-init](https://github.com/doitintl/kube-secrets-init) instead of Kubernetes secrets though.

**Pull request closed without merging**